examples/vision/reptile.py

```diff
@@ -0,0 +1,248 @@
+            )
+            ds = ds.batch(1)
+        # Iterate over the dataset to get each individual image and accompanying class
+        # and put that data into a dictionary.
+        self.data = {}
+
+        def extraction(image, label):
+            # This function will shrink the omniglot images into the desired size,
+            # scale the pixel values and convert the RGB image to graysclae.
+            image = tf.image.convert_image_dtype(image, tf.float32)
+            image = tf.image.rgb_to_grayscale(image)
+            image = tf.image.resize(image, [28, 28])
+            return image, label
+
+        for image, label in ds.map(extraction):
+            image = image.numpy()[0]
+            label = str(label.numpy()[0])
+            if label not in self.data:
+                self.data[label] = []
+            self.data[label].append(image)
+            self.labels = list(self.data.keys())
+
+    def get_mini_dataset(
+        self, batch_size, repetitions, shots, num_classes, split=False
+    ):
+
+        temp_labels = np.zeros(shape=(num_classes * shots))
+        temp_images = np.zeros(shape=(num_classes * shots, 28, 28, 1))
+        if split:
+            test_labels = np.zeros(shape=(num_classes))
+            test_images = np.zeros(shape=(num_classes, 28, 28, 1))
+        # Get a random subset of labels from the entire label set.
+        label_subset = random.choices(self.labels, k=num_classes)
+        for class_idx, class_obj in enumerate(label_subset):
+            # Use enumerated index value as a temporary label for mini_batch in
+            # few shot learning.
+            temp_labels[class_idx * shots : (class_idx + 1) * shots] = class_idx
+            # If creating a split dataset for testing, select an extra sample from each
+            # label to create the test dataset.
+            if split:
+                test_labels[class_idx] = class_idx
+                images_to_split = random.choices(
+                    self.data[label_subset[class_idx]], k=shots + 1
+                )
+                test_images[class_idx] = images_to_split[-1]
+                temp_images[
+                    class_idx * shots : (class_idx + 1) * shots
+                ] = images_to_split[:-1]
+            else:
+                # For each index in the randomly selected label_subset, sample the
+                # necessary number of images.
+                temp_images[
+                    class_idx * shots : (class_idx + 1) * shots
+                ] = random.choices(self.data[label_subset[class_idx]], k=shots)
+
+        dataset = tf.data.Dataset.from_tensor_slices(
+            (temp_images.astype(np.float32), temp_labels.astype(np.int32))
+        )
+        dataset = dataset.shuffle(100).batch(batch_size).repeat(repetitions)
+        if split:
+            return dataset, test_images, test_labels
+        return dataset
+
+
+import urllib3
+
+urllib3.disable_warnings()
```

Comment:
Why?

Reply:
Running it in Colab I get the following warnings 

/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)

I wasn't sure how to approach resolving these issues for someone who wants to quickly replicate the results.

Reply:
OK, please add a quick comment to explain that line.

Reply:
ok